plugin/package.json

```diff
@@ -24,25 +24,25 @@
     "react-native",
     "config-plugin"
   ],
-  "jest": {
-    "preset": "expo-module-scripts"
-  },
   "author": "Jakob Heuser <jakob@codedrift.com>",
   "license": "MIT",
   "bugs": {
     "url": "https://github.com/jakobo/expo-community-flipper/issues"
   },
   "homepage": "https://github.com/jakobo/expo-community-flipper#readme",
   "devDependencies": {
-    "@expo/config-types": "^46.0.1",
     "@types/semver": "^7.3.11",
-    "expo-module-scripts": "^2.0.0",
+    "expo": ">47.0.0-beta",
+    "expo-module-scripts": "^2.1.1",
+    "jest": "^29.2.2",
     "tslint": "^6.1.3",
-    "typescript": "^4.5.2"
+    "typescript": "^4.8.4"
   },
   "dependencies": {
-    "@expo/config-plugins": "^5.0.0",
-    "semver": "^7.3.7",
+    "semver": "^7.3.8",
     "ts-dedent": "^2.2.0"
+  },
+  "peerDependencies": {
+    "expo": ">=47.0.0"
```

Comment:
according to [the doc](https://docs.expo.dev/guides/config-plugins/#installing-dependencies), may also add the `peerDependenciesMeta` to prevent npm v7 installing the package for peer deps.

Reply:
I think that's the behavior we want, looking at https://github.blog/2021-02-02-npm-7-is-now-generally-available/#peer-dependencies

> npm 7 will block installations if an upstream dependency conflict is present that cannot be automatically resolved

For plugin v47.x I think expo 47+ in your dependency chain is a requirement. Otherwise, we may grab the wrong expo config plugin utilities with unpredictable results. Previously, when we depended on `@expo/config-plugins` we could just pin to the config plugin version we required but that versioning was outside of the expo SDK lifecycle.

That said, it may be more correct to make the peerDependency `47.x` which allows all released versions of expo 47 SDK or the more relaxed `>=47.0.0-beta.0 <48` to include all beta SDK releases. In either case, this would requires us to update the plugin for SDK 48 explicitly when the time comes.

Reply:
that sounds like a different mindset toward this. when we proposing the `peerDependenciesMeta` as well as the `"peerDependencies": { "expo": ">=46.0.0" }`, we though the community config-plugin could support multiple expo sdks without spending effort to follow each sdks updates. for example, a `expo-community-another-plugin@1.0.0` could support every sdks. the plugin authors don't have to publish new versions for each expo sdks, and the plugin users also don't have to update the plugin over again.

based on the status quo, the `expo-community-flipper`'s versioning convention is to follow expo sdk versioning, without peerDependenciesMeta and having `"peerDependencies": { "expo": "47.x" }` makes sense to me.

Reply:
I would love to not recertify the plugin with every SDK, but because we use a dangerous mod (regex), we're dependent on the contents of

https://github.com/expo/expo/blob/main/templates/expo-template-bare-minimum/ios/Podfile

Which is the base file used during prebuild IIRC. Since this file revs with versions of `react-native` and we tend to have a new `react-native` in an SDK release, it felt like the easiest thing to pin the version to. Definitely open to thoughts on what else we could use in its place to say "compatible with podfiles from X".

Reply:
I'd love to do this for 48. I'm in the expo discord (jakobox#7777), so we can talk more there. With build property support, we can simplify Flipper integrations down to a single pair of configuration values in the official plugin. We'd also bring the Podfile one step closer to the default Podfile from react-native.

And even better, this plugin will have served its purpose bridging the gap to a proper expo config plugin. 🎉 

```json5
{
  "expo": {
    "plugins": [
      [
        "expo-build-properties",
        {
          "android": {
            "flipper": true // or a flipper version string such as "0.128.0"
          },
          "ios": {
            "flipper": true // or a flipper version string such as "0.128.0"
          }
        }
      ]
    ]
  }
}
```

This also puts the option right next to the `use_frameworks` config, where we can remind folks in the docs that flipper & use_frameworks cannot be enabled at the same time (for as long as that issue persists on the rn/flipper side).